Dockerfile

```diff
@@ -11,39 +35,77 @@ RUN mix local.hex --force && \
     mix local.rebar --force
 
 # set build ENV
-ENV MIX_ENV=prod
+ENV MIX_ENV="prod"
 
 # install mix dependencies
 COPY mix.exs mix.lock ./
-COPY config config
-COPY assets/package.json assets/package-lock.json ./assets/
+RUN mix deps.get --only $MIX_ENV
+RUN mkdir config
 
-RUN mix do setup, deps.compile
+# copy compile-time config files before we compile dependencies
+# to ensure any relevant config change will trigger the dependencies
+# to be re-compiled.
+COPY config/config.exs config/lessons.exs config/redirects.exs config/${MIX_ENV}.exs config/
+RUN mix deps.compile
+
+COPY priv priv
 
 COPY lib lib
+
 COPY assets assets
-COPY priv priv
+
 COPY Makefile Makefile
 
+# install npm dependencies
+RUN cd assets && npm ci
+
+# install content from remote repository
 RUN make content
-RUN mix do assets.deploy, compile
+
+# generate rss and sitemap static files
+RUN mix school_house.gen.rss ${DEPLOY_DOMAIN}
+RUN mix school_house.gen.sitemap ${DEPLOY_DOMAIN}
+
+# compile assets
+RUN mix assets.deploy
+
+# Compile the release
+RUN mix compile
+
+# Changes to config/runtime.exs don't require recompiling the code
+COPY config/releases.exs config/
+
+COPY rel rel
 RUN mix release
 
-# prepare release image
-FROM alpine:3.9 AS app
+# start a new build stage so that the final image will only contain
+# the compiled release and other runtime necessities
+FROM ${RUNNER_IMAGE}
 
-RUN apk add --no-cache openssl ncurses-libs libstdc++
+RUN apt-get update -y && apt-get install -y libstdc++6 openssl libncurses5 locales \
+  && apt-get clean && rm -f /var/lib/apt/lists/*_*
 
-WORKDIR /app
+# Set the locale
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && locale-gen
+
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+WORKDIR "/app"
+RUN chown nobody /app
 
-RUN chown nobody:nobody /app
+# set runner ENV
+ENV MIX_ENV="prod"
 
-USER nobody:nobody
+# Appended by flyctl
+ENV ECTO_IPV6=true
+ENV ERL_AFLAGS="-proto_dist inet6_tcp"
 
-COPY --from=build --chown=nobody:nobody /app/_build/prod/rel/school_house ./
+# Only copy the final release from the build stage
+COPY --from=builder --chown=nobody:root /app/_build/${MIX_ENV}/rel/school_house ./
+COPY --from=builder --chown=nobody:root /app/priv/static ./static
```

Comment:
I'm copying the built static files and having fly host them on their cdn. This means that the Elixir app doesn't have to serve those and can gain even more speed.